src/client/array/dc_array.c

```diff
@@ -2068,18 +2068,24 @@ free_set_size_cb(tse_task_t *task, void *data)
 }
 
 static int
-punch_extent(daos_handle_t oh, daos_handle_t th, daos_size_t dkey_val, daos_off_t record_i,
-	     daos_size_t num_records, tse_task_t *task, d_list_t *task_list)
+punch_dkey_or_extent(daos_handle_t oh, daos_handle_t th, daos_size_t dkey_val, daos_off_t record_i,
```

Comment:
I think we should change record_i to start_offset of what we are punching and pass 0 and record_i + 1 rather than adding 1 internally.  Would be more clear and we don't have to rely on overflow

Reply:
https://github.com/daos-stack/daos/pull/14660 if you want to merge it in

Reply:
LGTM, thanks